components/DBTableMocker.php

```diff
@@ -27,28 +34,36 @@ public function getTableDescription(string $tableName):array {
 
             $query      = new Query("DESCRIBE `$tableName`");
 
-            $result  = $connection->query($query->getQueryString());
-            $columns = $result->fetch_all(MYSQLI_ASSOC);
+            $result        = $connection->query($query->getQueryString());
+            $columns       = $result->fetch_all(MYSQLI_ASSOC);
+            
+            foreach ($columns as $column) {
+                $this->columns[$column["Field"]] = $column;
+            }
 
             return $columns;
         }
 
         /**
-         * Sets the current column
+         * Sets the column which is being modified
          *
-         * @param string $column
-         * @return void
+         * @param string $columnName
+         * @return self
          */
-        public function column(string $column):void {
-            $this->currentColumn = $column;
+        public function column(string $columnName):self {
+            if (array_key_exists($columnName, $this->columns)) {
+                $this->currentColumn = $this->columns[$columnName];
+            }
+
+            return $this;
```

Comment:
Based on the column name we store the current column it the column with such name exists.